customer-terminate-service/domain/src/test/java/se/inera/intyg/cts/domain/repository/InMemoryCertificateRepository.java

```diff
@@ -0,0 +1,32 @@
+package se.inera.intyg.cts.domain.repository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import se.inera.intyg.cts.domain.model.Certificate;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationId;
+
+public class InMemoryCertificateRepository implements CertificateRepository {
```

Comment:
Reason why I didn't use mocking is that I don't want the tests to be too coupled to implementation. Instead by providing simple inMemory implementation of the repositories to remove any dependencies to databases or external systems the unit tests can easily add needed testdata and fokus on verifying the behaviour.

I think Mockito is great and have used it extensively, but it is not uncommon that wiring up the tests have to deal many implementation details which easily break when you refactor. I'm trying this approach instead, when writing the code from scratch, and aiming for high testability. If it shows that the approach doesn't work, which I don't see right now, I will rewrite the tests with mocks.

Reply:
I do not see how a Mocked interface is coupled to the implementation of the actual class? If the interface is changed the mocked values should be updated. Writing our own implementations of the classes and not using well tested and adopted test frameworks does not follow Best Practice as far as i know. Using a mock framework is vital if we would like to adopt test driven development. I do not see it as a sustainable solution to use these classes instead of Mocks. I can't say that i see the gain of testability when CertificateRepository is used in 1 class + unit test and this solution adds 33 lines of untested code. I have also used Mockito and other mocking frameworks extensively both in Java, JS and C#. My experiens is that the open source frameworks are the ones that works in the long run. (Partly due to the fact that the evolve with the programming language.) I would love to see what we can come up with using java 17 and the latest frameworks in case of high testability. :)